### PR TITLE
Fixes part of #935: Optimised test cases for HDPI devices ( Packages M-P)

### LINKED ISSUE #935: Optimise app module test cases
Currently all `app` module test cases pass for high-end devices like Pixel but we need to optimise these test cases to make sure that they pass on HDPI devices also, preferably `Nexus S - emulator`.

Except `player` package in test cases all test cases should pass on `HDPI` device.

When you create a new emulator device make sure that these three animations are turned off in developer options:
Screenshot 2020-04-02 at 11 38 17 AM

### PR DESCRIPTION
**Pull request by NullByte08**
## Explanation
Partially fixes #935
Tests made compatible for `HDPI` devices for packages `mydownloads` to `profileprogress`. All the tests pass successfully.

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/app/profile/PinPasswordActivityTest.kt [hunk 1]
```diff
@@ -528,6 +542,31 @@ class PinPasswordActivityTest {
     }
   }
 
+  fun getCurrentActivity(): Activity? {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/profile/PinPasswordActivityTest.kt]: Make sure you always solve all the possible mentioned warnings/suggestions by android studio.
```
private fun getCurrentActivity(): Activity? {
    var currentActivity: Activity? = null
    getInstrumentation().runOnMainSync {
      run {
        currentActivity = ActivityLifecycleMonitorRegistry.getInstance().getActivitiesInStage(
          Stage.RESUMED
        ).elementAtOrNull(0)
      }
    }
    return currentActivity
  }

  private inline fun <reified T : Activity> isVisible(): Boolean {
    val am =
      InstrumentationRegistry.getInstrumentation().targetContext.getSystemService(ACTIVITY_SERVICE) as ActivityManager
    val visibleActivityName = this.getCurrentActivity()!!::class.java.name
    return visibleActivityName == T::class.java.name
  }

  private inline fun <reified T : Activity> waitUntilActivityVisible() {
    val startTime = System.currentTimeMillis()
    while (!isVisible<T>()) {
      Thread.sleep(CONDITION_CHECK_INTERVAL)
      if (System.currentTimeMillis() - startTime >= TIMEOUT) {
        throw AssertionError("Activity ${T::class.java.simpleName} not visible after $TIMEOUT milliseconds")
      }
    }
  }
```


Also, you can create top level constants like this:
```
private const val TIMEOUT = 1000L
private const val CONDITION_CHECK_INTERVAL = 100L

@RunWith(AndroidJUnit4::class)
class PinPasswordActivityTest {
```

File: app/src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest.kt [hunk 2]
```diff
@@ -339,6 +345,33 @@ class ProfileChooserFragmentTest {
     }
   }
 
+  fun getCurrentActivity(): Activity? {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest.kt]: Ditto as above.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Looks great, the implementation is nice but you can optimise the code.

**Comment by NullByte08** [comment]
@rt4914 Changes done.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by rt4914** [comment]
@NullByte08 Merge this PR with latest branch so that `non_flaky_tests` can pass.

**Comment by NullByte08** [comment]
@rt4914 My branch is up-to-date with oppia/develop. But still, the non_flaky_tests are failing. Please have a look. Thanks.

**Comment by rt4914** [comment]
> @rt4914 My branch is up-to-date with oppia/develop. But still, the non_flaky_tests are failing. Please have a look. Thanks.

Tests are failing because of an import missing in `PinPasswordActivityTest`. Once you fix this, it should pass.

**Comment by NullByte08** [comment]
@rt4914 Changes done, please merge. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**